pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionFactory.java

```diff
@@ -254,15 +274,15 @@ public static AggregationFunction getAggregationFunction(FunctionContext functio
                 "ARRAY_AGG expects the 2nd argument to be literal, got: %s. The function can be used as "
                     + "arrayAgg(dataColumn, 'dataType', ['isDistinct'])", dataTypeExp.getType());
             DataType dataType = DataType.valueOf(dataTypeExp.getLiteral().getStringValue().toUpperCase());
-            boolean isDistinct = false;
+            boolean isDistinctArrayAgg = false;
```

Comment:
(minor) You may revert this change if you put braces around the `LISTAGG` case. I feel it is easier to read to keep these variables local to the case